internal/api/metadata/describe.go

```diff
@@ -14,7 +16,24 @@ type DescribeResponse struct {
 
 // Describe all endpoints on the server.
 func HandleDescribe(request *DescribeRequest) (*DescribeResponse, *core.APIError) {
-	response := DescribeResponse{core.GetAPIDescription()}
+	apiDescription := core.GetAPIDescription()
+
+	var err error
+	if apiDescription == nil || request.ForceUpdate {
+		routes := core.GetAPIRoutes()
+		if len(routes) == 0 {
+			return nil, core.NewInternalError("-501", request, "Unable to describe API endpoints when the cached routes are empty.")
+		}
+
+		apiDescription, err = core.Describe(routes)
+		if err != nil {
+			return nil, core.NewInternalError("-502", request, "Failed to describe API endpoints.").Err(err)
+		}
+
+		core.SetAPIDescription(apiDescription)
+	}
```

Comment:
I don't think the caching should be done like this.
You are still always forcing a high-cost operation on the first call.
Instead, I think you should just pull the description from the resources dir unless someone explicitly asked for a full computation (or the resources is missing).